### PR TITLE
Adds change PIN form

### LINKED ISSUE #149: Change my pin for proxy / paid accounts
Split out of #47 

`/user/patron/changeMyPin`

A separate feature that is needed. After going through reset/request PIN workflow, we need to provide a form that will do the actual changes.

### PR DESCRIPTION
**Pull request by mejackreed**
Fixes #149 and thus finishes the resetPIN / change PIN workflow.

To test this, you must have an account setup that is linked to your email. `JACKPIN` account is currently set to my email. Happy to pair up and show reviewers the whole flow.

resetpin

### CODE REVIEW

File: app/controllers/reset_pins_controller.rb [hunk 1]
```diff
@@ -6,9 +6,38 @@ class ResetPinsController < ApplicationController
...
+  private
+
+  def change_pin_with_token_params
+    params.require(%I[token pin])
+  end
+
+  ##
+  # Hacky workaround Rails routing helpers to have the <> not encoded
+  def change_pin_with_token_unencoded_url
+    change_pin_with_token_url('foo').gsub('foo', '<RESET_PIN_TOKEN>')
```
Comment (cbeer) [inline app/controllers/reset_pins_controller.rb]: Where is `RESET_PIN_TOKEN` coming from? Symphony land?

Comment (mejackreed) [reply app/controllers/reset_pins_controller.rb]: Yep. Symphony looks for the string `<RESET_PIN_TOKEN>` and injects the actual token in a URL crafted in the email sent to the user.

File: config/routes.rb [hunk 2]
```diff
@@ -21,6 +21,8 @@
 
   get '/reset_pin', to: 'reset_pins#index'
   post '/reset_pin', to: 'reset_pins#reset'
+  get '/change_pin/:token', to: 'reset_pins#change_form', as: :change_pin_with_token
```
Comment (mejackreed) [inline config/routes.rb]: Open to other options here on how we craft a URL with a token in it. Tried to keep it simple.

File: app/views/reset_pins/change_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<h1>Change PIN</h1>
+<%= form_tag change_pin_path, method: :post do %>
+  <%= hidden_field_tag :token, params['token'] %>
```
Comment (mejackreed) [inline app/views/reset_pins/change_form.html.erb]: This comes from the URL that is visited. Symphony when using the `changeMyPin` action, will verify that the correct token came through.

File: app/services/symphony_client.rb [hunk 4]
```diff
@@ -73,15 +73,24 @@ def fines(patron_key)
...
     JSON.parse(response.body)
   end
 
+  def change_pin(token, pin)
+    response = request('/user/patron/changeMyPin', method: :post, json: {
+      resetPinToken: token,
+      newPin: pin
+    })
+
+    response
```
Comment (mejackreed) [inline app/services/symphony_client.rb]: We need the status code so no JSON parsing here.

File: app/controllers/reset_pins_controller.rb [hunk 5]
```diff
@@ -6,9 +6,38 @@ class ResetPinsController < ApplicationController
 
   def index; end
 
+  def change_form; end
```
Comment (mejackreed) [inline app/controllers/reset_pins_controller.rb]: Names are hard? /shrug open to other suggestions here.

File: config/routes.rb (hunk 2 above)
Comment (mejackreed) [reply config/routes.rb]: Also, we could possibly implement  (future)`change_pin_with_pin` more of a regular "change my password" type workflow

File: app/views/reset_pins/change_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  <div>
+    Your new PIN must:
+    <ul>
+      <li>be 6-25 characters long</li>
```
Comment (mejackreed) [inline app/views/reset_pins/change_form.html.erb]: These rules aren't actually validated anywhere, but its more of a current feature parity thing. We could add future validation if we need to.

File: app/controllers/reset_pins_controller.rb (hunk 1 above)
Comment (cbeer) [reply app/controllers/reset_pins_controller.rb]: Got it; would you mind documenting that in the method comment?

Comment (mejackreed) [reply app/controllers/reset_pins_controller.rb]: Sure thing.. added.

**Comment by mejackreed** [comment]
I'm also 👎 to code climate here.. not really a useful thing. Happy to dismiss, but what's our policy?

**Pull request merged**